prince-of-versions/src/test/java/co/infinum/princeofversions/tests/PrinceOfVersionsTest.java

```diff
@@ -334,4 +339,275 @@ public UpdateConfigLoader newInstance() {
         Mockito.verify(callback, Mockito.times(0)).onNoUpdate(ArgumentMatchers.<String, String>anyMap());
     }
 
-}
+    @Test
+    public void testCheckingWhenUpdateShouldBeMade() throws PackageManager.NameNotFoundException {
+        Context context = setupContext("2.0.0");
+        PrinceOfVersions updater = new PrinceOfVersions(context, provider, repository, setupSdkInt(16));
+        updater.checkForUpdates(new LoaderFactory() {
+            @Override
+            public UpdateConfigLoader newInstance() {
+                return new ResourceFileLoader("valid_update_no_sdk_values.json");
+            }
+        }, callback);
+
+        Mockito.verify(callback, Mockito.times(1))
+                .onNewUpdate(eq("2.4.5"), Mockito.anyBoolean(), ArgumentMatchers.<String, String>anyMap());
```

Comment:
I do not think we should use anyBoolean(). It is important if an update is mandatory or not.